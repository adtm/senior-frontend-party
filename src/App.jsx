import React, { Fragment } from "react";
import { Router, Switch, Route } from "react-router-dom";
import history from "./utils/history";

import Auth from "./utils/Auth";
import Login from "./pages/Login";
import List from "./pages/List";
import GlobalStyle from "./utils/globalStyles";

import ErrorPage from "./pages/Error";

const App = () => (
  <Auth>
    {({ isAuthenticated, authenticate, logout }) => (
      <Router history={history}>
        <Fragment>
          <GlobalStyle />
          <Switch>
            <Route
              exact
              path="/"
              render={() => <Login authenticate={authenticate} />}
            />
            {isAuthenticated ? (
              <Route path="/list" render={() => <List logout={logout} />} />
            ) : null}
            <Route component={ErrorPage} />
          </Switch>
        </Fragment>
      </Router>
    )}
  </Auth>
);

export default App;
