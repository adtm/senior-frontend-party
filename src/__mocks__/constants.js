exports.MOCK_TOKEN = "super-secret";
