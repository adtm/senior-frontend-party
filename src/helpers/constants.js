exports.USERNAME_ERROR_MSG = "Username is required!";
exports.PASSWORD_ERROR_MSG = "Password is required!";
