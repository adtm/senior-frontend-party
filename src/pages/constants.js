exports.SUCCESS_COLOR = "#9fd533";
exports.WARNING_COLOR = "red";
exports.LOGIN_TEXT = "Log In";

exports.DEFAULT_USERNAME = "tesonet";
exports.DEFAULT_PASSWORD = "partyanimal";
