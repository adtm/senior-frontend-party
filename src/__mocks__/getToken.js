import { MOCK_TOKEN } from "./constants";

jest.mock("../api/api");
import { getToken } from "../api/api";

getToken.mockImplementation(
  () => new Promise(resolve => resolve({ token: MOCK_TOKEN }))
);
