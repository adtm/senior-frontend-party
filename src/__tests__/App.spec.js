import React from "react";
import {
  render,
  fireEvent,
  cleanup,
  waitForElement,
} from "react-testing-library";

import "../__mocks__/getToken";
import App from "../App";

import { MOCK_TOKEN } from "../__mocks__/constants";

beforeEach(() => localStorage.clear());
afterEach(cleanup);

describe("Local Storage", () => {
  it("should write and clean token to LS after login & logout", async () => {
    const { getByText, container } = render(<App />);

    expect(container.firstChild).toMatchSnapshot();
    expect(localStorage.getItem("token")).toEqual(null);

    fireEvent.click(getByText(/log in/i));

    await waitForElement(() => getByText(/Logout/i));
    expect(container.firstChild).toMatchSnapshot();
    expect(localStorage.getItem("token")).toEqual(MOCK_TOKEN);

    fireEvent.click(getByText(/Logout/i));

    expect(localStorage.getItem("token")).toEqual(null);
    expect(container.firstChild).toMatchSnapshot();
  });
});
