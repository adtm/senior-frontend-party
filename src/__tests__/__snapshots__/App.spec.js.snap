// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Local Storage should write and clean token to LS after login & logout 1`] = `
<div
  class="d-flex sc-ifAKCX eiBayV"
>
  <div
    class="col-sm-12 col-md-6 col-lg-4 m-auto container"
  >
    <div
      class="sc-bxivhb hUgcLF"
    >
      <img
        class="sc-htpNat eUlSDF"
        src="white-logo.png"
      />
    </div>
    <form
      class=""
    >
      <div
        class="position-relative form-group"
      >
        <div
          class="input-group"
        >
          <div
            class="input-group-prepend"
            style="border: 0px; background: rgb(255, 255, 255); border-top-left-radius: 5px; border-bottom-left-radius: 5px; padding-left: 15px;"
          >
            <span
              class="input-group-text"
              style="border: 0px; color: rgb(178, 178, 178);"
            >
              <img
                class="sc-bdVaJa jyHzQx"
                src="ico-username.svg"
              />
            </span>
          </div>
          <input
            class="form-control-lg form-control"
            name="username"
            placeholder="Username"
            style="border: 0px; color: rgb(178, 178, 178); height: 56px; line-height: 1.875; border-radius: 0; border-top-right-radius: 5px; border-bottom-right-radius: 5px;"
            type="text"
            value="tesonet"
          />
        </div>
      </div>
      <div
        class="position-relative form-group"
      >
        <div
          class="input-group"
        >
          <div
            class="input-group-prepend"
            style="border: 0px; background: rgb(255, 255, 255); border-top-left-radius: 5px; border-bottom-left-radius: 5px; padding-left: 15px;"
          >
            <span
              class="input-group-text"
              style="border: 0px; color: rgb(178, 178, 178);"
            >
              <img
                class="sc-bdVaJa jyHzQx"
                src="ico-lock.svg"
              />
            </span>
          </div>
          <input
            class="form-control-lg form-control"
            name="password"
            placeholder="Password"
            style="border: 0px; color: rgb(178, 178, 178); height: 56px; line-height: 1.875; border-radius: 0; border-top-right-radius: 5px; border-bottom-right-radius: 5px;"
            type="password"
            value="partyanimal"
          />
        </div>
      </div>
      <button
        class="w-100 sc-bwzfXH fPDeGH btn btn-#9fd533"
        type="submit"
      >
        Log In
      </button>
    </form>
  </div>
</div>
`;

exports[`Local Storage should write and clean token to LS after login & logout 2`] = `
<div
  class="sc-gzVnrw jHLeDV"
>
  <img
    alt="testio."
    class="sc-htoDjs ivXlwq"
    src="black-logo.png"
  />
  <button
    class="sc-iwsKbI iUDrjP"
  >
    <img
      alt="logout-button"
      class="sc-dnqmqq jzveCj"
      src="ico-logout.svg"
    />
    <span>
      Logout
    </span>
  </button>
</div>
`;

exports[`Local Storage should write and clean token to LS after login & logout 3`] = `
<div
  class="d-flex sc-ifAKCX eiBayV"
>
  <div
    class="col-sm-12 col-md-6 col-lg-4 m-auto container"
  >
    <div
      class="sc-bxivhb hUgcLF"
    >
      <img
        class="sc-htpNat eUlSDF"
        src="white-logo.png"
      />
    </div>
    <form
      class=""
    >
      <div
        class="position-relative form-group"
      >
        <div
          class="input-group"
        >
          <div
            class="input-group-prepend"
            style="border: 0px; background: rgb(255, 255, 255); border-top-left-radius: 5px; border-bottom-left-radius: 5px; padding-left: 15px;"
          >
            <span
              class="input-group-text"
              style="border: 0px; color: rgb(178, 178, 178);"
            >
              <img
                class="sc-bdVaJa jyHzQx"
                src="ico-username.svg"
              />
            </span>
          </div>
          <input
            class="form-control-lg form-control"
            name="username"
            placeholder="Username"
            style="border: 0px; color: rgb(178, 178, 178); height: 56px; line-height: 1.875; border-radius: 0; border-top-right-radius: 5px; border-bottom-right-radius: 5px;"
            type="text"
            value="tesonet"
          />
        </div>
      </div>
      <div
        class="position-relative form-group"
      >
        <div
          class="input-group"
        >
          <div
            class="input-group-prepend"
            style="border: 0px; background: rgb(255, 255, 255); border-top-left-radius: 5px; border-bottom-left-radius: 5px; padding-left: 15px;"
          >
            <span
              class="input-group-text"
              style="border: 0px; color: rgb(178, 178, 178);"
            >
              <img
                class="sc-bdVaJa jyHzQx"
                src="ico-lock.svg"
              />
            </span>
          </div>
          <input
            class="form-control-lg form-control"
            name="password"
            placeholder="Password"
            style="border: 0px; color: rgb(178, 178, 178); height: 56px; line-height: 1.875; border-radius: 0; border-top-right-radius: 5px; border-bottom-right-radius: 5px;"
            type="password"
            value="partyanimal"
          />
        </div>
      </div>
      <button
        class="w-100 sc-bwzfXH fPDeGH btn btn-#9fd533"
        type="submit"
      >
        Log In
      </button>
    </form>
  </div>
</div>
`;
